### DISCUSSION TITLE
How to align center the Add 2 Calendar Button?

### DISCUSSION THREAD

**Discussion by BrettHoebel** [add2cal/add-to-calendar-button#374] [Q&A] [answered]
I am unable to align center the Add 2 Calendar button and I don't see any short code on how to do that in the installation guide. Any suggestions on how to align the Add 2 Calendar button in Word Press Divi would be appreciated.

**Comment by jekuer** [comment, marked as answer]
Hi there,

The button is rendered as block element. This makes it block the full line, while the visible element flows to the left.
You can change this behavior by setting the attribute `inline` or `inline="true"`. This makes it render inline-block instead. You are then able to control it a little better and usually this is enough to properly align it.

If you are still experiencing problems, you should check the surrounding container. If possible, assign it the `display:flex;` via css. You can then even position the block style element in a more detailled manner (for example via `display: flex; justify-content: center;`).

All the best!

**Reply by naturedamends** [reply]
If the column is short - it will compress the drop down menu and become unusable 

![image](https://github.com/add2cal/add-to-calendar-button/assets/120284608/c48b9920-12df-4439-8d0e-7192ba55cabf)

**Reply by jekuer** [reply]
Can you share a screenshot of the full element (incl. the button) and table?
Which version are you running?
What settings are you using for the button?

**Reply by naturedamends** [reply]
```
<script src="https://cdn.jsdelivr.net/npm/add-to-calendar-button@2" async defer></script>
```
```
              <add-to-calendar-button id="add-to-calendar-component" name=""
                description="" startDate="2024-02-23" startTime="10:15"
                endTime="17:45" timeZone="Europe/London" location=""
                options="'Apple','Google','iCal','Outlook.com','Yahoo'" hideRichData 
                hideTextLabelButton></add-to-calendar-button>
```
```
          <div class="row p-0 m-2 m-0">
            <div class="col-8 ">
            </div>
            <div class="col-4 p-2">
           its here
         </div>
      </div>
```

In that picture it works fine, as you can see the grid is large enought. But If i made the bootstrap div a col-2, it reduces the size of the dropdown like seen in the picture

**Reply by jekuer** [reply]
There are multiple ways to solve this smoothly, but this would require optimization of your source code.
The rather "brute force" method could be the forceOverlay option.

**Reply by holyl** [reply]
Hi, I have tried the suggestions above, but I need help. The button is still aligned to the left, while all the other elements in the column are centered. Please could you advise me on the correct CSS class to address to align the button to the center of the column: https://mistrovstvi2024.adkcr.cz/? Than you very much in advance.

**Reply by jekuer** [reply]
@holyl , try not to use the `inline` option, but wrap the element in another container, which you then could center. OR you could add `style="display:block"` to the <add-to-calendar-button> tag (not sure how you are using it as this would not be easily possible with WordPress shortcode or Block).

**Reply by holyl** [reply]
Hi I tried both, blick and shortcode. As for now, I´m using the Block. I have added the suggested style tag but no effect so far. Do you have any other ideas to try? Vielen Dank :-)

**Reply by jekuer** [reply]
When using WP Blocks:

1. click the button.
2. click the 3 dots.
3. hit "group"
4. on the right, "transform to row"
5. at "JUSTIFICATION", center it

![image](https://github.com/add2cal/add-to-calendar-button/assets/8572883/d72a4cbf-58aa-4561-8019-e18678401bb6)

**Reply by holyl** [reply]
Thank you very much for this proposal, unfortunately, this doesn't work either. In the editing window, the position is shown correctly but on the live page the button still sticks to the left. PLZ do you have any other card in your sleeve? BG THX in advance.

**Reply by jekuer** [reply]
Checking your source code, WordPress is not adding the style, which it should when using the settings from the screenshot above. You can try to change those settings back and forth. Maybe there is some weird caching issue. Or another plugin mixing things up.

You could also try the shortcode block, put it into a row and center. Maybe this works.

Alternatively, you could use the HTML block and do something like this:

```
<div style="display:flex;width:100%;justify-content:center;">
  <add-to-calendar-button name="Mistrovství a Klání" description="Play with me!" startdate="2024-06-15" starttime="08:00" endtime="18:00" timezone="Europe/Prague" location="Vimperk" options="'Apple','Google','iCal','Outlook.com'" buttonstyle="round" size="3" label="Přidat do kalendáře" lightmode="bodyScheme" language="cs"></add-to-calendar-button>
</div>
```

**Comment by holyl** [comment]
Hi Jens,
The thing with the HTML code worked finally. Herzlichen Dank for you effort and the outstanding support you´ve provided to me!

**Reply by jekuer** [reply]
You're welcome
